python/helpers.py

```diff
@@ -0,0 +1,172 @@
+                        }}
+                    }}
+                ]
+            }},
+            "properties": {{
+                "insurance": {{
+                    "bsonType": "object",
+                    "properties": {{
+                        "policyNumber": {{
+                            "encrypt": {{
+                                "bsonType": "int",
+                                "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic"
+                            }}
+                        }}
+                    }}
+                }},
+                "medicalRecords": {{
+                    "encrypt": {{
+                        "bsonType": "array",
+                        "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Random"
+                    }}
+                }},
+                "bloodType": {{
+                    "encrypt": {{
+                        "bsonType": "string",
+                        "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Random"
+                    }}
+                }},
+                "ssn": {{
+                    "encrypt": {{
+                        "bsonType": "int",
+                        "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic"
+                    }}
+                }}
+            }}
+        }}
+    }}
+    """.format(data_key=data_key)
+
+    return json_util.loads(patient_schema)
+
+
+class ClientBuilder:
+    def __init__(self,
+                 kms_providers=None,
+                 key_alt_name="demo-data-key",
+                 key_db="encryption",
+                 key_coll="__keyVault",
+                 schema=None,
+                 connection_string="mongodb://localhost:27017",
+                 mongocryptd_bypass_spawn=False,
+                 mongocryptd_spawn_path="mongocryptd"):
+
+        super().__init__()
+        if kms_providers is None:
+            raise ValueError("kms_provider is required")
+        self.kms_providers = kms_providers
+        self.key_alt_name = key_alt_name
+        self.key_db = key_db
+        self.key_coll = key_coll
+        self.key_vault_namespace = f"{key_db}.{key_coll}"
+        self.schema = schema
+        self.connection_string = connection_string
+        self.mongocryptd_bypass_spawn = mongocryptd_bypass_spawn
+        self.mongocryptd_spawn_path = mongocryptd_spawn_path
+
+        fle_opts = AutoEncryptionOpts(
+            self.kms_providers,
+            self.key_vault_namespace,
+            mongocryptd_bypass_spawn=self.mongocryptd_bypass_spawn,
+            mongocryptd_spawn_path=self.mongocryptd_spawn_path)
+
+        self.encrypted_client = MongoClient(
+            self.connection_string, auto_encryption_opts=fle_opts)
+
+        self.regular_client = MongoClient(self.connection_string)
+
+        self.key_vault = self.encrypted_client.get_database(
+            self.key_db).get_collection(self.key_coll)
+
+        # clients are required to create a unique sparse index on keyAltNames
```

Comment:
Is sparse a requirement? The `getKeyVault()` shell method did not specify sparse, and I think you are not specifying it either.